### PR TITLE
Add secret mounter

### PR DESCRIPTION
**Pull request by diptadas**
```
kloader run -c mymap -m /tmp/kloader/ -b pwd --k8s-config ~/.kube/config  -v=20
kloader run -s mysecret -m /tmp/kloader/ -b pwd --k8s-config ~/.kube/config  -v=20
```

### CODE REVIEW

File: moutSecret.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
+package main
```
Comment (sadlil) [inline moutSecret.go]: same convention for file name.

File: run.go [hunk 2]
```diff
@@ -13,8 +13,13 @@ func newRunCmd() *cobra.Command {
 		Use:   "run",
 		Short: "Run and hold kloader",
 		Run: func(cmd *cobra.Command, args []string) {
-			mounter := NewConfigMapMounter(getRestConfig(), configMap, mountDir, bashFile)
-			mounter.Run()
+			if configMap != "" {
+				mounter := NewConfigMapMounter(getRestConfig(), configMap, mountDir, bashFile)
+				mounter.Run()
+			} else if secret != "" {
```
Comment (sadlil) [inline run.go]: If's a separate condition. They are not either or.

File: run.go [hunk 3]
```diff
@@ -23,8 +28,16 @@ func newRunCmd() *cobra.Command {
...
-		log.Fatal("ConfigMap/MountDir is required, but not provided")
+	if configMap == "" && secret == "" {
+		log.Fatal("ConfigMap/Secret is required, but not provided")
+	}
+
+	if configMap != "" && secret != "" {
+		log.Fatal("Either ConfigMap or Secret is required, but both are provided")
+	}
+
+	if mountDir == "" {
```
Comment (sadlil) [inline run.go]: what if i need seperate mount dir for secrets and config maps.

File: main.go [hunk 4]
```diff
@@ -40,6 +40,7 @@ func main() {
 	rootCmd.AddCommand(version.NewCmdVersion())
 
 	rootCmd.PersistentFlags().StringVarP(&configMap, "config-map", "c", "", "Configmap name that needs to be mount")
+	rootCmd.PersistentFlags().StringVarP(&secret, "secret", "s", "", "Secret name that needs to be mount")
```
Comment (sadlil) [inline main.go]: what about multiple secrets?

**Review by sadlil — CHANGES_REQUESTED** [review]

File: run.go (hunk 2 above)
Comment (tamalsaha) [reply run.go]: I told him to make it either or. I think that makes sense, since the actual command is going to be same.

File: main.go (hunk 4 above)
Comment (tamalsaha) [reply main.go]: @sadlil , What do you mean? What is the use-case?

**Pull request merged**